akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -349,12 +440,79 @@ class CircuitBreaker(
   private def resetBreaker(): Unit = transition(HalfOpen, Closed)
 
   /**
+   * Invokes all onSuccess callback handlers.
+   *
+   * @param start time in nanoseconds of call invocation
+   */
+  private def notifyCallSuccessListeners(start: Long): Unit = if (!successListeners.isEmpty) {
+    val elapsed = System.nanoTime() - start
+    val iterator = successListeners.iterator()
+    while (iterator.hasNext) {
+      val listener = iterator.next()
+      executor.execute(new Runnable {
+        def run() = listener.accept(elapsed)
+      })
```

Comment:
Is it important that the listeners are executed in parallell, or could we create one `Runnable` that iterates over the listeners and call them one by one, to avoid the overhead of creating and adding one `Runnable` for each listener?